textproto/multipart.go

```diff
@@ -272,15 +272,14 @@ func (r *MultipartReader) NextPart() (*Part, error) {
 			return bp, nil
 		}
 
-		if r.isFinalBoundary(line) {
+		// Only return EOF if we've reached the final boundary that we are looking for, and are not expecting any
+		// further parts to the multipart encoding. This prevents the parser from stopping too soon during nested
+		// multipart boundaries https://www.rfc-editor.org/rfc/rfc2046#section-5.1
+		if r.isFinalBoundary(line) && !expectNewPart {
```

Comment:
seems like this is worth PRing to upstream`go-message`. I think they're receptive, especially for a bug fix

Reply:
I've tried to PR upstream on `go-imap` before, and the same maintainer just ghosted me. I can try again.

Reply:
https://github.com/emersion/go-message/pull/162